code/__DEFINES/icon_smoothing.dm

```diff
@@ -26,7 +26,7 @@ DEFINE_BITFIELD(smoothing_flags, list(
 
 #define QUEUE_SMOOTH(thing_to_queue) if(thing_to_queue.smoothing_flags & (SMOOTH_CORNERS|SMOOTH_BITMASK)) {SSicon_smooth.add_to_queue(thing_to_queue)}
 
-#define QUEUE_SMOOTH_NEIGHBORS(thing_to_queue) for(var/neighbor in orange(1, thing_to_queue)) {var/atom/atom_neighbor = neighbor; QUEUE_SMOOTH(atom_neighbor)}
+#define QUEUE_SMOOTH_NEIGHBORS(thing_to_queue) for(var/atom/neighbor as anything in orange(1, thing_to_queue)) {if(neighbor.smooth) QUEUE_SMOOTH(neighbor)}
```

Comment:
What the heck is going on here

Reply:
not sure if i can use an if statement here but it is basically ```for(var/atom/neighbor as() in orange(1,A))
		if(neighbor.smooth)
			queue_smooth(neighbor)```

Reply:
why? that's not consistent with our current behavior

Reply:
you can by the way, though if you want newlines you need to use {} and ;

Reply:
reverted